iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -607,41 +600,12 @@ await _internalRetryPolicy
             }
         }
 
-        public override Task OpenAsync(CancellationToken cancellationToken)
+        public override async Task OpenAsync(CancellationToken cancellationToken)
         {
-            return EnsureOpenedAsync(true, cancellationToken);
+            await OpenAsync(cancellationToken, true).ConfigureAwait(false);
```

Comment:
By convention, cancellation tokens always go last in the parameter list.

Reply:
Fixed in latest commit
https://github.com/Azure/azure-iot-sdk-csharp/pull/2632/commits/47c9ce49828b188a309c818f8486485e2a38adbe